### PR TITLE
Enh #29: AutoStart Tour for new Users

### PR DESCRIPTION
**Pull request by yuriimaz**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [ ] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [x] All tests are passing
- [ ] New/updated tests are included
- [ ] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
https://github.com/humhub/humhub-internal/issues/29

### CODE REVIEW

File: protected/humhub/modules/user/components/User.php [hunk 1]
```diff
@@ -147,6 +147,17 @@ public function afterLogin($identity, $cookieBased, $duration)
         parent::afterLogin($identity, $cookieBased, $duration);
     }
 
+    /**
+     * Set updated_by for current user
+     *
+     * @param $identity
+     * @return void
+     */
+    public function selfSetUpdatedBy($identity)
```
Comment (luke-) [inline protected/humhub/modules/user/components/User.php:156]: I would like to keep this component as compact as possible. And prefer instead, two private methods:

- tour\Module::getIsFirstLogin(User $user): bool
- tour\Module::setFirstLoginDone(User $user): void

**Review by luke- — CHANGES_REQUESTED** [review]
@yuriimaz Please see my comment and 

- Would be great to have an option to disable this autostart. e.g. via `tour\Module::$autoStartOnNewAccounts=false`.
- This feature should be covered by acceptance tests
 
Thanks!

**Review by luke- — CHANGES_REQUESTED** [review]
Thanks. Can you please check my remarks and add some tests?

File: protected/humhub/modules/tour/Module.php [hunk 2]
```diff
@@ -24,18 +24,62 @@ class Module extends \humhub\components\Module
      */
     public $isCoreModule = true;
 
+    public $autoStartOnNewAccounts = false;
```
[the remark this answers is not in the document]
Comment (yuriimaz) [reply protected/humhub/modules/tour/Module.php:31]: Done

File: protected/humhub/modules/tour/Module.php [hunk 3]
```diff
@@ -24,18 +24,62 @@ class Module extends \humhub\components\Module
...
      * Event Callback
      */
     public static function onDashboardSidebarInit($event)
     {
-        if (Yii::$app->user->isGuest)
+        if (Yii::$app->user->isGuest) {
             return;
+        }
+
+        $module = Yii::$app->getModule('tour');
```
[the remark this answers is not in the document]
Comment (yuriimaz) [reply protected/humhub/modules/tour/Module.php:80]: Done

File: protected/humhub/modules/tour/Module.php [hunk 4]
```diff
@@ -24,18 +24,71 @@ class Module extends \humhub\components\Module
      */
     public $isCoreModule = true;
 
+    /**
+     * @author yuriimaz
```
Comment (luke-) [inline protected/humhub/modules/tour/Module.php:28]: We do not use `@author` tags within the files. Can you please remove.

File: protected/humhub/modules/tour/tests/codeception/acceptance/AutoTourCest.php [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) 2018 HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\tour\tests\codeception\acceptance;
+
+use tour\AcceptanceTester;
+
+class AutoTourCest
```
Comment (luke-) [inline protected/humhub/modules/tour/tests/codeception/acceptance/AutoTourCest.php:12]: Interesting would be cases that:

- The Tour is automatically triggered on the first login and not shown again on the second login.
- Is not triggered on the first login if disabled via Module::autoStartOnNewAccounts

**Review by luke- — CHANGES_REQUESTED** [review]

File: protected/humhub/modules/tour/tests/codeception/acceptance/AutoTourCest.php (hunk 5 above)
Comment (yuriimaz) [reply protected/humhub/modules/tour/tests/codeception/acceptance/AutoTourCest.php:14]: I've made the appropriate changes for the accetnas test, but unfortunately I'm currently running MacOS on an arm64 and can't run either selenium or phantomjs to test it, as they aren't yet optimized for this processor.

File: protected/humhub/modules/tour/Module.php [hunk 6]
```diff
@@ -24,18 +24,71 @@ class Module extends \humhub\components\Module
...
 
+    /**
+     * @var bool enable auto-start tour for new accounts
+     * @since 1.15
+     */
+    public $autoStartOnNewAccounts = false;
+
+    /**
+     * Check is first login
+     * @author yuriimaz
```
Comment (luke-) [inline protected/humhub/modules/tour/Module.php:35]: Please check  author tags also on other locations

**Review by luke- — CHANGES_REQUESTED** [review]

File: protected/humhub/modules/tour/Module.php (hunk 6 above)
Comment (yuriimaz) [reply protected/humhub/modules/tour/Module.php]: Done

**Comment by luke-** [comment]
@yuriimaz Can you please review the failing tests?

**Comment by yuriimaz** [comment]
Accepatnce tes for Auto Tour fixed

**Review by luke- — APPROVED** [review]

**Pull request merged**